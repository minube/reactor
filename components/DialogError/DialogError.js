import { number, oneOfType, string } from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import { Icon, Text } from '..';
import styles from './DialogError.style';

const DialogError = ({ icon, title, ...inherit }) => (
  <View style={[styles.container, inherit.style]}>
    <View style={[styles.content]}>
      <Text body level={2} style={styles.text}>{title}</Text>
      {icon && <Icon style={styles.icon} value={icon} />}
    </View>
    <View style={[styles.arrow]} />
  </View>
);

DialogError.propTypes = {
  icon: oneOfType([number, string]),
  title: string.isRequired,
};

DialogError.defaultProps = {
  icon: undefined,
};

export default DialogError;
