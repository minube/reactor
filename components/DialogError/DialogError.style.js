import { StyleSheet } from 'react-native';
import { LAYOUT, THEME } from '../../common';

const {
  BORDER_RADIUS, COLOR, SPACE, UNIT,
} = THEME;

export default StyleSheet.create({
  arrow: {
    borderLeftColor: COLOR.TRANSPARENT,
    borderLeftWidth: SPACE.XS,
    borderTopColor: COLOR.LIGHT_ERROR,
    borderTopWidth: SPACE.XS,
    borderRightColor: COLOR.TRANSPARENT,
    borderRightWidth: SPACE.XS,
    left: UNIT * 2,
    position: 'absolute',
    top: UNIT * 4.4,
  },

  container: {
    marginBottom: SPACE.S,
  },

  content: {
    ...LAYOUT.STYLE.ROW,
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    paddingHorizontal: SPACE.S,
    paddingVertical: SPACE.REGULAR / 2,
    borderRadius: BORDER_RADIUS,
    backgroundColor: COLOR.LIGHT_ERROR,
    minWidth: UNIT * 6,
    minHeight: UNIT * 4.4,
    width: '100%',
  },

  icon: {
    marginLeft: SPACE.S,
  },

  text: {
    color: COLOR.ERROR,
  },
});
