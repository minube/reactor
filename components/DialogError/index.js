export { default } from './DialogError'; // eslint-disable-line
