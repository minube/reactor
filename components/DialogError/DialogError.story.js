import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, select } from '@storybook/addon-knobs/react';

import DialogError from './DialogError';
import MOCKS from '../Icon/Icon.mocks';

const TEXT = 'You must complete all fields';

storiesOf('☑️ DialogError', module)
  .add('default', () => (
    <DialogError title={TEXT}/>
  ))
  .add('icon', () => (
    <DialogError title={TEXT} icon="alertColor" />
  ))
  .add('🏀 Playground', () => (
    <DialogError
      title={text('currency', TEXT)}
      icon={select('value', MOCKS.ICONS, MOCKS.VALUE)}
    />
  ));
