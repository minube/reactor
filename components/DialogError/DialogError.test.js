import React from 'react';
import renderer from 'react-test-renderer';

import DialogError from './DialogError';
import MOCKS from '../Icon/Icon.mocks';

const TEXT = 'You must complete all fields'

describe('<DialogError>', () => {
  it('renders when {text}', () => {
    const tree = renderer.create(<DialogError title={TEXT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {icon}', () => {
    const tree = renderer.create(<DialogError title={TEXT} icon={MOCKS.VALUE_NAV.RIGHT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
