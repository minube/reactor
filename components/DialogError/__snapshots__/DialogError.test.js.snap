// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogError> renders when {text} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 12,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "rgba(239, 81, 96, 0.2)",
          "borderRadius": 3,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minHeight": 44,
          "minWidth": 60,
          "paddingHorizontal": 12,
          "paddingVertical": 10,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 14,
            "lineHeight": 24,
          },
          false,
          Object {
            "color": "#ef5160",
          },
        ]
      }
    >
      You must complete all fields
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "transparent",
          "borderLeftWidth": 8,
          "borderRightColor": "transparent",
          "borderRightWidth": 8,
          "borderTopColor": "rgba(239, 81, 96, 0.2)",
          "borderTopWidth": 8,
          "left": 20,
          "position": "absolute",
          "top": 44,
        },
      ]
    }
  />
</View>
`;

exports[`<DialogError> when {icon} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 12,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "rgba(239, 81, 96, 0.2)",
          "borderRadius": 3,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minHeight": 44,
          "minWidth": 60,
          "paddingHorizontal": 12,
          "paddingVertical": 10,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 14,
            "lineHeight": 24,
          },
          false,
          Object {
            "color": "#ef5160",
          },
        ]
      }
    >
      You must complete all fields
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../components/Icon/assets/nav-arrow.png",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 20,
            "width": 20,
          },
          Object {
            "marginLeft": 12,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "transparent",
          "borderLeftWidth": 8,
          "borderRightColor": "transparent",
          "borderRightWidth": 8,
          "borderTopColor": "rgba(239, 81, 96, 0.2)",
          "borderTopWidth": 8,
          "left": 20,
          "position": "absolute",
          "top": 44,
        },
      ]
    }
  />
</View>
`;
